packages/client/src/views/GrantActivity.vue

```diff
@@ -0,0 +1,167 @@
+<template>
+  <b-card
+    header-bg-variant="white"
+    footer-bg-variant="white"
+  >
+    <template #header>
+      <h3 class="m-2">
+        Grant Activity
+      </h3>
+    </template>
+    <div>
+      <div class="feature-text">
+        Stay up to date with this grant.
+        <b-icon
+          v-b-tooltip
+          class="ml-2"
+          title="Follow this grant to receive an email notification when others follow or leave a note."
+          icon="info-circle-fill"
+        />
+      </div>
+      <b-button
+        block
+        size="lg"
+        :variant="followBtnVariant"
+        class="mb-4"
+        data-follow-btn
+        :disabled="loadingFollowState"
+        @click="toggleFollowState"
+      >
+        <span class="h4">
+          <b-icon
+            icon="check-circle-fill"
+            class="mr-2"
+          />
+        </span>
+        <span class="h5">
+          {{ followBtnLabel }}
+        </span>
+      </b-button>
+      <div>
+        <span
+          v-if="showFollowSummary"
+          data-follow-summary
+        >{{ followSummaryText }}</span>
+        <span
+          v-if="showFollowSummary && showNotesSummary"
+          class="mx-1"
+        >&bull;</span>
+        <span v-if="showNotesSummary">{{ notesSummaryText }}</span>
+      </div>
+    </div>
+
+    <template #footer>
+      <!-- Feed -->
+    </template>
+  </b-card>
+</template>
+
+<script>
+import { mapActions, mapGetters } from 'vuex';
+
+export default {
+  components: {},
+  data() {
+    return {
+      userIsFollowing: false,
+      loadingFollowState: false,
```

Comment:
I'm not sure about the best way to resolve (perhaps initializing as `loadingFollowingState: true`, but open to suggestions), but I noticed that when testing in my local dev environment, the button initially defaults to the blue "Follow" state before the underlying API request completes. This makes me wonder if we can initialize to a loading/unknown state instead.

Testing flow that reveals what I noticed:
1. Given: Grant 1234 is not being followed by the current user
2. Go to the details page for Grant 1234. A blue "Follow" button displays.
3. Click "Follow". The button will remain blue and go into a disabled (greyed-out) state (as expected).
4. Once the underlying PUT request completes successfully, the button will turn green, say "Following", and return to enabled state.
5. Reload (or leave and return to) the details page for Grant 1234. Initially, the button shows as blue, enabled, and says "Follow" before eventually changing to green and "Following". 
    - This is easier to observe for testing by adding `await new Promise((cb) => setTimeout(cb, 2000));` at the start of the [GET `/:grantId/follow` request handler](https://github.com/usdigitalresponse/usdr-gost/blob/20f5dcad6390856a119e568a750913152cd8d836/packages/server/src/routes/grants.js#L508-L520) in `packages/server/src/routes/grants.js`.

Reply:
good catch, I've updated to be disabled until verified